cadence/contracts/services/AxelarGasService.cdc

```diff
@@ -0,0 +1,139 @@
+import IAxelarGasService from "../interfaces/IAxelarGasService.cdc"
+import FungibleToken from "fungible-token"
+import FlowToken from "flow-token"
+
+// Reverence Solidity Implementation: https://github.com/axelarnetwork/axelar-cgp-solidity/blob/main/contracts/gas-service/AxelarGasService.sol
+pub contract AxelarGasService: IAxelarGasService {
+    pub event NativeGasPaidForContractCall(
+        sourceAddress: Address,
+        destinationChain: String,
+        destinationAddress: String,
+        payloadHash: [UInt8],
+        gasFeeAmount: UInt256,
+        refundAddress: Address,
+    )
+
+    pub event NativeGasPaidForExpressCall(
+        sourceAddress: Address,
+        destinationChain: String,
+        destinationAddress: String,
+        payloadHash: [UInt8],
+        gasFeeAmount: UInt256,
+        refundAddress: Address,
+    )
+
+    pub event NativeGasAdded(
+        txHash: String,
+        logIndex: UInt256,
+        gasFeeAmount: UInt256,
+        refundAddress: Address,
+    )
+
+    pub event NativeExpressGasAdded(
+        txHash: String,
+        logIndex: UInt256,
+        gasFeeAmount: UInt256,
+        refundAddress: Address,
+    )
+
+    pub event Refund(
+        txHash: String,
+        logIndex: UInt256,
+        reciever: Address,
+        token: String, //potentially not needed
+        amount: UInt256,
+    )
+
+    access(all) fun payNativeGasForContractCall(
+        sender: Address,
```

Comment:
I dont think the sender value is too sensitive in this case.  These functions are mainly used for off chain accounting of gas payments, so if the sender value is wrong, then the funds will most likely just be incorrectly accounted.  I dont think it holds any secuirty risk, just up to the user to correctly add funds, but this will most likely be abstracted away with dapp interaction.